src/private/MDMTransaction+Private.h

```diff
@@ -18,9 +18,22 @@
 
 @protocol MDMPlan;
 
+// typedef for identifying what type of operation the MDMTransactionLog is associated with
+typedef NS_ENUM(NSInteger, MDMTransactionLogType) {
+  MDMTransactionLogTypeUncategorized = 0,
+  MDMTransactionLogTypeAddNamedPlan,
+  MDMTransactionLogTypeRemoveNamedPlan,
+};
+
 @interface MDMTransactionLog : NSObject
-@property(nonatomic) NSArray<id<MDMPlan>> *plans;
-@property(nonatomic) id target;
+
+@property(nonatomic, copy) NSArray<id<MDMPlan>> *plans;
+@property(nonatomic, strong) id target;
+@property(nonatomic, copy) NSString *name;
+@property(nonatomic, assign) MDMTransactionLogType transactionLogType;
```

Comment:
This might be representable as `BOOL isRemoval` if we use nullable state of `name` to indicate whether this is a named operation.